app/core/exceptions/exception.py

```
@@ -1,15 +1,23 @@
+from app.core.exceptions.error_code import ErrorCode
 
 
 class AppException(Exception):
     def __init__(
         self,
-        error_code: object,
+        error_code: ErrorCode,
         detail: Optional[str] = None,
         status_code: Optional[int] = None,
+        content: Optional[BeautifulSoup] = None,
+        data: Optional[Any] = None,
```

Ruff (0.14.7)

Dynamically typed expressions (typing.Any) are disallowed in `data`

(ANN401)